Common symbols not included on macOS

Building the post executable fails on macOS. You get undefined symbols for variables in modules that contain no functions. By passing `-c` to ranlib on macOS those symbols are included when linking.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,6 +37,12 @@ if(OPENMP)
   find_package(OpenMP REQUIRED COMPONENTS Fortran)
 endif()
 
+if(APPLE)
+  # The linker on macOS does not include `common symbols` by default
+  # Passing the -c flag includes them and fixes an error with undefined symbols
+  set(CMAKE_Fortran_ARCHIVE_FINISH "<CMAKE_RANLIB> -c <TARGET>")
+endif()
+
 find_package(bacio REQUIRED)
 find_package(crtm REQUIRED)
 find_package(g2 REQUIRED)

--- a/sorc/ncep_post.fd/CMakeLists.txt
+++ b/sorc/ncep_post.fd/CMakeLists.txt
@@ -158,7 +158,6 @@ list(APPEND EXE_SRC
     getVariable.f
     getlvls.f
     intio_tags.f
-    native_endianness.f
     retrieve_index.f
     wrf_io_flags.f
     wrf_io_flags.h)

